python/test/unit/runtime/test_subproc.py

```diff
@@ -71,3 +71,36 @@ def test_compile_in_forked_subproc() -> None:
     proc.start()
     proc.join()
     assert proc.exitcode == 0
+
+
+def compile_empty_kernel_with_gc(attrs):
+
+    @triton.jit
+    def empty_kernel():
+        pass
+
+    import gc
+    gc.collect()
+    src = ASTSource(fn=empty_kernel, signature={}, attrs=attrs, constants=dict())
+    triton.compile(src=src, target=target)
+
+
+def test_compile_in_forked_subproc_with_forced_gc() -> None:
+    reset_tmp_dir()
+    import gc
```

Comment:
Can we also add some comment for this test to indicate what's its testing? Given this is a very subtle issue; it's non trivial for folks reading the test code to understand what it is trying to do.